server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -422,6 +428,29 @@ public static void ensureNodeCommissioned(DiscoveryNode node, Metadata metadata)
         }
     }
 
+    public static void ensureRemoteStoreNodesCompatibility(DiscoveryNode node, Metadata metadata) {
+        assert metadata.settings()
+            .get("cluster.remote_store.segment.repository")
+            .equals(node.getAttributes().get("remote_store.segment.repository.name"));
+        assert metadata.settings()
+            .get("cluster.remote_store.translog.repository")
+            .equals(node.getAttributes().get("remote_store.translog.repository.name"));
+
+        RepositoriesMetadata repositories = metadata.custom(RepositoriesMetadata.TYPE);
+        if (repositories != null && node != null) {
+            for (RepositoryMetadata existingRepositoryMetadata : repositories.repositories()) {
+                if (existingRepositoryMetadata.name().equals(node.getAttributes().get("remote_store.segment.repository.name"))) {
+                    RepositoryMetadata newSegmentRepositoryMetadata = buildSegmentRepositoryMetadata(node);
+                    existingRepositoryMetadata.equalsIgnoreGenerations(newSegmentRepositoryMetadata);
+                }
+                if (existingRepositoryMetadata.name().equals(node.getAttributes().get("remote_store.translog.repository.name"))) {
+                    RepositoryMetadata newTranslogRepositoryMetadata = buildTranslogRepositoryMetadata(node);
+                    existingRepositoryMetadata.equalsIgnoreGenerations(newTranslogRepositoryMetadata);
```

Comment:
`equalsIgnoreGenerations` seems too strict as it is checking for all the settings to be equal. Ideally we should check only the settings which are read only .